aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -72,6 +79,23 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
 
     @VisibleForTesting
     internal lateinit var realPlugin: RealAWSCognitoAuthPlugin
+    private val queueFacade: KotlinAuthFacadeInternal by lazy {
+        KotlinAuthFacadeInternal(realPlugin)
+    }
+
+    private val queueChannel = Channel<Job>(capacity = Channel.UNLIMITED).apply {
+        GlobalScope.launch {
+            consumeEach { it.join() }
+        }
+    }
+
+    private fun Exception.toAuthException(): AuthException {
+        return if (this is AuthException) {
```

Comment:
Why not just call `CognitoExceptionConvertor.lookup` here? We can then use `toAuthException` everywhere else and it will help us clean up at other places too.

Reply:
Its not a service exception so all would end up converted to Unknown

Reply:
Unless you are proposing modifying the converter to first check if already an AuthException and if so, keep as is.